### PR TITLE
vpc: create vpc tiers with dns server of vpc instead of zone

### LINKED ISSUE #8598: VPC network tier DNS doesn't reflect VPC DNS settings
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
VPC, VR
~~~

##### CLOUDSTACK VERSION

~~~
4.18, 4.19 
~~~

##### SUMMARY

When creating a VPC a user adds DNS entries for the VPC, those DNS servers should be used by all network tiers of the VPC, but instead the network tiers still use the Zone

##### STEPS TO REPRODUCE

~~~
Create a new VPC
Enter DNS1 and DNS2 details
Create a network tier
Check DNS entries for the network tier
~~~

##### EXPECTED RESULTS

~~~
Network tiers should use VPC defined DNS entries
~~~

##### ACTUAL RESULTS

~~~
Network tiers use Zone defined DNS entries
~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #8598 

with this change
```
root@r-575-VM:~# cat /etc/dnsmasq.d/cloud.conf
dhcp-hostsfile=/etc/dhcphosts.txt
listen-address=127.0.0.1,172.17.1.1,172.17.2.1,172.17.3.1
dhcp-range=set:interface-eth2-0,172.17.1.1,static
dhcp-option=tag:interface-eth2-0,15,cs2cloud.internal
dhcp-option=tag:interface-eth2-0,6,172.17.1.1,10.0.32.1,8.8.8.8
dhcp-option=tag:interface-eth2-0,3,172.17.1.1
dhcp-option=eth2,26,1500
dhcp-option=tag:interface-eth2-0,1,255.255.255.0
dhcp-range=set:interface-eth3-1,172.17.2.1,static
dhcp-option=tag:interface-eth3-1,15,cs2cloud.internal
dhcp-option=tag:interface-eth3-1,6,172.17.2.1,8.8.8.8,1.1.1.1
dhcp-option=tag:interface-eth3-1,3,172.17.2.1
dhcp-option=eth3,26,1500
dhcp-option=tag:interface-eth3-1,1,255.255.255.0
dhcp-range=set:interface-eth4-2,172.17.3.1,static
dhcp-option=tag:interface-eth4-2,15,cs2cloud.internal
dhcp-option=tag:interface-eth4-2,6,172.17.3.1,8.8.8.8,1.1.1.1
dhcp-option=tag:interface-eth4-2,3,172.17.3.1
dhcp-option=eth4,26,1500
dhcp-option=tag:interface-eth4-2,1,255.255.255.0
root@r-575-VM:~#
```

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

File: server/src/main/java/com/cloud/network/NetworkModelImpl.java [hunk 1]
```diff
@@ -2723,6 +2733,12 @@ public Pair<String, String> getNetworkIp6Dns(final Network network, final DataCe
         if (StringUtils.isNotBlank(network.getIp6Dns1())) {
             return new Pair<>(network.getIp6Dns1(), network.getIp6Dns2());
         }
+        if (network.getVpcId() != null) {
+            Vpc vpc = vpcDao.findById(network.getVpcId());
+            if (vpc != null && StringUtils.isNotBlank(vpc.getIp6Dns1())) {
+                return new Pair<>(vpc.getIp6Dns1(), vpc.getIp6Dns2());
+            }
+        }
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/NetworkModelImpl.java:2741]: this block is implemented twice. can you extract to a method?

Comment (weizhouapache) [reply server/src/main/java/com/cloud/network/NetworkModelImpl.java:2741]: @DaanHoogland 
There is a small difference:
`ip4` vs `ip6`

File: server/src/main/java/com/cloud/network/NetworkModelImpl.java [hunk 2]
```diff
@@ -2715,6 +2719,12 @@ public Pair<String, String> getNetworkIp4Dns(final Network network, final DataCe
         if (StringUtils.isNotBlank(network.getDns1())) {
             return new Pair<>(network.getDns1(), network.getDns2());
         }
+        if (network.getVpcId() != null) {
+            Vpc vpc = vpcDao.findById(network.getVpcId());
```
[the remark this answers is not in the document]
Comment (weizhouapache) [reply server/src/main/java/com/cloud/network/NetworkModelImpl.java:2723]: this is similar as the network DNSs (the Dns2 can be null)
```
        if (StringUtils.isNotBlank(network.getDns1())) {
            return new Pair<>(network.getDns1(), network.getDns2());
        }
```

**Review by vishesh92 — APPROVED** [review]
clgtm

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
this has been merged into main branch as part of #7919
it has been verified

**Pull request merged**